sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SpanDescriptionExtractor.java

```diff
@@ -62,4 +66,117 @@ private OtelSpanInfo descriptionForDbSystem(final @NotNull ReadableSpan otelSpan
+
+    final @Nullable String httpMethod = attributes.get(SemanticAttributes.HTTP_METHOD);
+    if (httpMethod != null) {
+      return descriptionForHttpMethod(otelSpan, httpMethod);
+    }
+
+    final @Nullable String httpRequestMethod =
+        attributes.get(SemanticAttributes.HTTP_REQUEST_METHOD);
+    if (httpRequestMethod != null) {
+      return descriptionForHttpMethod(otelSpan, httpRequestMethod);
+    }
+
+    final @Nullable String dbSystem = attributes.get(SemanticAttributes.DB_SYSTEM);
+    if (dbSystem != null) {
+      return descriptionForDbSystem(otelSpan);
+    }
+
+    return new OtelSpanInfo(name, name, TransactionNameSource.CUSTOM);
+  }
+
+  @SuppressWarnings("deprecation")
+  private OtelSpanInfo descriptionForHttpMethod(
+      final @NotNull SpanData otelSpan, final @NotNull String httpMethod) {
+    final @NotNull String name = otelSpan.getName();
+    final @NotNull SpanKind kind = otelSpan.getKind();
+    final @NotNull StringBuilder opBuilder = new StringBuilder("http");
+    final @NotNull Attributes attributes = otelSpan.getAttributes();
+    final @NotNull Map<String, Object> dataFields = new HashMap<>();
+    dataFields.put("http.request.method", httpMethod);
+
+    if (SpanKind.CLIENT.equals(kind)) {
+      opBuilder.append(".client");
+    } else if (SpanKind.SERVER.equals(kind)) {
+      opBuilder.append(".server");
+    }
+    final @Nullable String httpTarget = attributes.get(SemanticAttributes.HTTP_TARGET);
+    final @Nullable String httpRoute = attributes.get(SemanticAttributes.HTTP_ROUTE);
+    @Nullable String httpPath = httpRoute;
+    if (httpPath == null) {
+      httpPath = httpTarget;
+    }
+    final @NotNull String op = opBuilder.toString();
+
+    final @Nullable Long httpStatusCode =
+        attributes.get(SemanticAttributes.HTTP_RESPONSE_STATUS_CODE);
+    if (httpStatusCode != null) {
+      dataFields.put("http.response.status_code", httpStatusCode);
+    }
+
+    final @Nullable String serverAddress = attributes.get(SemanticAttributes.SERVER_ADDRESS);
+    if (serverAddress != null) {
+      dataFields.put("server.address", serverAddress);
+    }
+
+    final @Nullable String urlFull = attributes.get(SemanticAttributes.URL_FULL);
+    if (urlFull != null) {
+      dataFields.put("url.full", urlFull);
+      if (httpPath == null) {
+        httpPath = urlFull;
+      }
+    }
+
+    if (httpPath == null) {
+      return new OtelSpanInfo(op, name, TransactionNameSource.CUSTOM, dataFields);
+    }
+
+    final @NotNull String description = httpMethod + " " + httpPath;
+    final @NotNull TransactionNameSource transactionNameSource =
+        httpRoute != null ? TransactionNameSource.ROUTE : TransactionNameSource.URL;
+
+    return new OtelSpanInfo(op, description, transactionNameSource, dataFields);
+  }
+
+  @SuppressWarnings("deprecation")
+  private OtelSpanInfo descriptionForDbSystem(final @NotNull SpanData otelSpan) {
```

Comment:
Couldn't we have the overloaded method receiving `ReadableSpan` call this method with `ReadableSpan.toSpanData()`?